bin/analysis_submission.py

```diff
@@ -209,54 +174,94 @@ def upload_to_ENA(self, trialcount):
+            self.save_accession(analysis_accession)
+            print('> Analysis ID: {}'.format(analysis_accession))
+            return analysis_accession
+        elif receipt_attributes.get('success') == 'false':  # Retry submission if unsuccessful
+            time.sleep(10)
+            attempts += 1
+            if attempts > 3:
+                analysis_accession = '> ERROR - Submission failed for {}: {}'.format(webin_loc, error.decode())
+                print(analysis_accession)
+                return analysis_accession
+            else:
+                self.submission(attempts)
+
+    def retrieve_json_info(self, output, error, attempts, webin_loc):
+        """
+        Handle information from JSON output following the submission
+        :param output: Output requests object from submission
+        :param error: Error output requests object from submission
+        :param attempts: Number of attempts for submission
+        :return: Submission accession or error message
+        """
+        output = json.loads(output)         # Convert JSON into dictionary object
+        try:
+            submission_id = output['submissionId']
+            self.save_accession(submission_id)
+            print('> Submission ID: {}'.format(submission_id))
+            return submission_id
+        except:
+            # If the output could not obtained, retry the submission
+            time.sleep(10)
+            attempts += 1
+            if attempts > 3:
+                submission_id = '> ERROR - Submission failed for {}: {}'.format(webin_loc, error.decode())
+                print(submission_id)
+                return submission_id
+            else:
+                self.submission(attempts)
 
     def submission(self, attempts):
         """
         Carry out the submission
         :param attempts: The number of times the submission has been attempted
         """
-        submission_loc = os.path.join(self.parent_dir,
-                                      'submission')  # Prefix for the name of the submission XML with file path
-        analysis_loc = os.path.join(self.parent_dir,
-                                    'analysis')  # Prefix for the name of the analysis XML with file path
+        webin_loc = os.path.join(self.parent_dir, 'webin')          # Prefix for the name of the Webin XML with file path
+
+        # Construct the appropriate submission command and run
         if self.test is True:
-            command = 'curl -u {}:{} -F "SUBMISSION=@{}_{}.xml" -F "ANALYSIS=@{}_{}.xml" "https://wwwdev.ebi.ac.uk/ena/submit/drop-box/submit/"'.format(
-                self.analysis_username, self.analysis_password, submission_loc, self.datestamp, analysis_loc,
-                self.datestamp)
+            command = 'curl -u {}:{} -X POST -H "accept: */*"  -H "Content-Type: multipart/form-data" -F "file=@{}_{}.xml;type=text/xml" "https://wwwdev.ebi.ac.uk/ena/submit/webin-v2/{}"'.format(
```

Comment:
The inclusion of the `-as` flag made it seem like this script would support both submission options, but from looking at the code, it looks like a direct replacement? We should support both endpoint options. I would recommend either:

1. remove the `-as` flag and make v2 a different branch in git (in which case you should not be making this PR to merge directly into `main`) or
2. keep the `-as` flag and make this script able to work with both versions of the submission service

I don't have a strong feeling either way - whichever you prefer. But the current form where you include a `-as` flag that doesn't do anything isn't ideal.

Reply:
Yes that's correct, so the tool supports both endpoints in V2, the `-as` flag controls whether a user wants to use the `submit/queue` version, as by default it goes with a synchronous option (submit) - which is the equivalent to Webin V1.